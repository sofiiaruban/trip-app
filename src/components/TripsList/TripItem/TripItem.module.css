.trip {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  border: 1px solid rgba(128, 128, 128, 0.2);
  flex-basis: auto;
  width: 200px;
  &.active {
    border: 2px solid rgb(16, 16, 211);
  }
}
.tripInfo {
  padding: 0.5rem;
}
.cityImg {
  width: 100%;
  max-height: 15rem;
  object-fit: fit;
}
.cityName {
  font-weight: 600;
  font-size: 24px;
  margin: 0;
}
