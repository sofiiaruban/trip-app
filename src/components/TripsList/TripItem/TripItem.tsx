import { FC } from 'react'
import styles from './TripItem.module.css'
import { TripProps } from '../types'

const TripItem: FC<TripProps> = ({
  cityImgSrc,
  cityName,
  startDate,
  endDate
}) => {
  const formattedStartDate = startDate.toLocaleDateString()
  const formattedEndDate = endDate.toLocaleDateString()

  return (
    <article className={styles.trip}>
      <img
        src={cityImgSrc}
        className={styles.cityImg}
        alt={`${cityName} city`}
      />
      <div className={styles.tripInfo}>
        <p className={styles.cityName}>{cityName}</p>
        <p className={styles.tripDates}>
          <span>{formattedStartDate}</span> - <span>{formattedEndDate}</span>
        </p>
      </div>
    </article>
  )
}

export default TripItem
