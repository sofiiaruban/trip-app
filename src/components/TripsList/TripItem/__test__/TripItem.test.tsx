import { render } from '@testing-library/react'
import TripItem from '../TripItem'

test('TripItem matches snapshot', () => {
  const setIsActiveMock = jest.fn()
  const commonProps = {
    setIsActive: setIsActiveMock,
    cityImgSrc: '/src',
    cityName: 'KYIV',
    startDate: new Date('24-10-01'),
    endDate: new Date('24-10-03')
  }

  const { container } = render(<TripItem {...commonProps} />)

  expect(container).toMatchSnapshot()
})
