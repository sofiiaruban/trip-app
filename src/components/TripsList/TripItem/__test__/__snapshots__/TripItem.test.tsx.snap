// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripItem matches snapshot 1`] = `
<div>
  <article
    class="trip"
  >
    <img
      alt="KYIV city"
      class="cityImg"
      src="/src"
    />
    <div
      class="tripInfo"
    >
      <p
        class="cityName"
      >
        KYIV
      </p>
      <p
        class="tripDates"
      >
        <span>
          Invalid Date
        </span>
         - 
        <span>
          Invalid Date
        </span>
      </p>
    </div>
  </article>
</div>
`;
